app/services/LegacyIdentityDbUserService.scala

```diff
@@ -27,7 +27,7 @@ class LegacyIdentityDbUserService(val dbConfig: DatabaseConfig[JdbcProfile])(imp
 
   def fetchUserByIdentityId(id: String): Future[Option[LegacyUser]] =
     db.run(
-      sql"SELECT id, okta_id, braze_uuid, jdoc->'publicFields'->'username', jdoc->'consents' FROM users WHERE id = $id LIMIT 1"
+      sql"SELECT id, okta_id, braze_uuid, jdoc->'publicFields'->>'username', jdoc->'consents' FROM users WHERE id = $id LIMIT 1"
```

Comment:
`->` gets the field as a Json object , `-->` gets the field as text.}
 see https://www.postgresql.org/docs/9.5/functions-json.html